packages/components/src/components/checkbox-wrapper/checkbox-wrapper-styles.ts

```diff
@@ -1,100 +1,105 @@
         '&(input)': {
-          position: 'static',
-          width: size,
-          height: size,
+          position: 'relative',
+          width: '28px',
+          height: '28px',
           flexShrink: 0,
           display: 'block',
           margin: 0,
           padding: 0,
           WebkitAppearance: 'none', // iOS safari
           appearance: 'none',
           boxSizing: 'border-box',
-          backgroundSize: size,
-          backgroundPosition: hasVisibleState ? '-2px -2px' : '-1px -1px',
-          backgroundColor,
+          backgroundSize: '24px',
+          backgroundColor: 'transparent',
           transition: ['border-color', 'background-color'].map(getTransition).join(),
-          opacity: 1,
-          border: hasVisibleState ? `2px solid ${formStateColor}` : `1px solid ${contrastMediumColor}`,
-          borderRadius: 0,
-          outline: '1px solid transparent',
-          outlineOffset: '2px',
-          cursor: 'pointer',
+          border: `2px solid ${uncheckedColor}`,
+          borderRadius: borderRadiusSmall,
+          outline: 0,
+          cursor: isDisabled ? 'not-allowed' : 'pointer',
         },
         '&(input:checked)': {
-          backgroundImage: `url('data:image/svg+xml;charset=UTF-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="${iconColor}" d="M9 19l-6-7h1.5l4.49 5.36L19.5 5H21L9 19z"/></svg>')`,
+          borderColor: checkedColor,
+          backgroundColor: checkedColor,
+          backgroundImage: getInlineSVG(
+            `<path fill="${checkedIconColor}" d="m20.22,7.47l-1.47-1.42-9.26,9.02-4.24-4.15-1.47,1.42,5.71,5.6,10.73-10.47Z"/>`
+          ),
         },
         '&(input:indeterminate)': {
-          backgroundImage: `url('data:image/svg+xml;charset=UTF-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="${iconColor}" d="M3 11h18v1H3z"/></svg>')`,
-        },
-        '&(input:checked), &(input:indeterminate)': {
-          borderColor: formStateColor || contrastHighColor,
-          backgroundColor: formStateColor || contrastHighColor,
+          backgroundImage: getInlineSVG(`<path fill="${indeterminateIconColor}" d="m20,11v2H4v-2h16Z"/>`),
         },
-        ...hoverMediaQuery({
-          '&(input:not(:disabled):hover), .label:hover ~ &(input:not(:disabled))': {
-            borderColor: formStateHoverColor || primaryColor,
+        ...(!isDisabled && {
+          ...hoverMediaQuery({
+            '&(input:hover), .text:hover ~ &(input)': {
+              borderColor: uncheckedHoverColor,
+            },
+            '&(input:checked:hover), .text:hover ~ &(input:checked)': {
+              borderColor: checkedHoverColor,
+              backgroundColor: checkedHoverColor,
+            },
+          }),
+          '&(input:focus)::before': {
+            content: '""',
+            position: 'absolute',
+            ...getInsetJssStyle(-2),
+            border: `${borderWidthBase} solid ${focusColor}`,
```

Comment:
Due that the focus indicator in non-checked state is based on color change only (from grey outline to focus outline) the contrast ratios are too low which can harm accessibility, especially if we look at upcoming version of WCAG 2.2.

Recommendation: use same offset and color like in checked state.

See more information here: https://www.sarasoueidan.com/blog/focus-indicators/#new-focus-indicator-accessibility-requirements-in-wcag-2.2

Reply:
That's something you need to discuss in general with @marcelbertram, @neuncreative and @muellernaumann in general and shouldn't block this PR.